### ISSUE TITLE
Hashtag search results missing in the "hashtags" tab

### ISSUE THREAD

**Issue by mgrzeca** [VernissageApp/VernissageWeb#135]
When I search for posts using the hashtag #photography, relevant posts appear correctly under the "statuses" tab. However, the matching hashtags don't show up in the "hashtags" tab, even though there are posts using this tag. 

It’s also possible that I don’t fully understand the intended purpose of the "hashtags" tab and how it’s supposed to function :)

**Comment by mczachurski** [comment]
The hashtag search is based on “Trending Hashtags” (Yearly). This list currently stores a maximum of 1,000 of the most popular hashtags from the past year, sorted by popularity. It is refreshed in the background every hour for optimization purposes. Additionally, this allows for an ordered display of results in the list. If certain hashtags do not appear in search results, this may indicate that either the task has not yet added these tags to the list or that they did not make it into the top 1,000 most popular hashtags (currently, vernissage.photos does not yet have this volume of tags).

It’s also important to note that hashtag search is designed to locate tags containing a specific word, rather than posts. Once an appropriate hashtag is found, clicking on it will display posts that include this tag. I hope this helps clarify the functionality a bit 😊.

**Comment by mgrzeca** [comment]
But the problem with trending hashtags is that it only applies to local statuses, and posts from other instances don’t make it into the trends, right? Am I understanding this correctly?

**Comment by mczachurski** [comment]
Trending hashtags list stores all tags regardless of whether the post is added locally or fetched from a remote server. The local server has no knowledge of tags that exist on other instances; the only source of information is the statuses fetched (or added) on the local server.

**Comment by mczachurski** [comment]
I found the source of the problem. I made an incorrect assumption and based tag popularity on a similar concept as status popularity, meaning a status is more popular if it has more likes. However, tag popularity should depend on how frequently a particular tag appears in statuses within the system (regardless of whether someone liked the status or not). I also increased the list to 10,000 (for all popular statuses, users, and tags). All changes done in the API.
